templates/devshells/haskell/flake.nix

```diff
@@ -0,0 +1,39 @@
+{
```

Comment:
> I came up with a basic shell for Haskell recently, and it looks [like this](https://github.com/aspiwack/haskell-benign/blob/99bc379ac14177cb5345c6fc96d13b2996163f72/shell.nix) (you can ignore just, and probably git). It is not perfect. For once it doesn't guarantee that the installed Ormolu version and the version linked in the HLS is the same (maybe I should use the direct invocation of HLS, rather than the provided wrapper, I'm not sure what the difference is). In fact, maybe we shouldn't tie the Ormolu version with the GHC version at all! (also the devtool bit is to avoid downloading HLS on CI because it's more than 1GB of closure… but this should be trivial to achieve in Nickel shells, hopefully).

@aspiwack (and others if you are interested) for what it's worth, in Zeus we have done the following: https://github.com/tweag/zeuslogics/blob/master/nix/default.nix#L62-L76
i.e. use an overlay to retrieve the Ormolu package used by HLS (otherwise we had inconsistent formatting between formatting with HLS and formatting in the CL/`pre-commit-hooks.nix`).

Courtesy of @amesgen for the trick

Reply:
This hurts my brain a little. But thanks for the trick! This sounds quite useful.

Reply:
> it doesn't guarantee that the installed Ormolu version and the version linked in the HLS is the same

In which situation can this happen? My understanding was that each `haskellPackages` set had only one version for each package (barring custom overrides), so `haskell.packages.{blah}.ormolu` and `haskell.packages.{blah}.haskell-language-server` should use the same Ormolu version, right?

Reply:
Aaah indeed, so “barring custom overrides” is not just a theoretical concern. Thanks for the pointer.

> Partially related, starting in [HLS 2.3.0.0](https://github.com/haskell/haskell-language-server/releases/tag/2.3.0.0), I added the possibility to configure HLS to use an external ormolu binary via a configuration flag. For more information, see https://github.com/haskell/haskell-language-server/pull/3771

I've seen that, yes! Hopefully we'll just use that in the future (but the default HLS in Nixpkgs is still 1.something, so it's a bit early unfortunately)

Reply:
> but the default HLS in Nixpkgs is still 1.something, so it's a bit early unfortunately

Which `nixpkgs` branch are you referring to? On `nixos-unstable` the current version is 2.3.0.0+, that's what we use for Zeus.